src/main/java/spoon/support/template/SubstitutionVisitor.java

```diff
@@ -194,11 +191,7 @@ private String substituteInDocComment(String docComment) {
 					it.remove();
 				}
 			}
-			for (CtTypeMember typeMember : ctClass.getTypeMembers()) {
-				if (!(typeMember instanceof CtField)) {
-					continue;
-				}
-				CtField<?> field = (CtField<?>) typeMember;
+			for (CtField<?> field : new ArrayList<>(ctClass.getFields())) { // copy is required for removing while iterating
```

Comment:
`getFields` already returns a new List but why do you iterate on `getFields` and no more `getTypeMembers`?